src/main/scala/viper/server/utility/AstGenerator.scala

```diff
@@ -35,10 +38,21 @@ class AstGenerator(private val _logger: Logger,
       throw new NoSuchFileException(vpr_file_path)
     }
     
-    val args: Seq[String] = Seq(vpr_file_path)
     _logger.info(s"Parsing `$vpr_file_path` ...")
-
-    _frontend.execute(args)
+    var filteredArgs: Seq[String] = Seq(vpr_file_path)
+    for (option <- optionWhiteList) {
+      val optionArgIndex = argList.indexOf(option)
+      if (optionArgIndex != -1) {
+        filteredArgs = argList.slice(optionArgIndex, 2) ++ filteredArgs
+      }
+    }
+    for (flag <- flagWhiteList) {
+      val flagArgIndex = argList.indexOf(flag)
+      if (flagArgIndex != -1) {
+        filteredArgs = Seq(argList(flagArgIndex)) ++ filteredArgs
+      }
+    }
+    _frontend.execute(filteredArgs)
```

Comment:
what would happen if you pass the entire `argList` to `_frontend`?

Reply:
Some of them may be backend-specific, in which case argument parsing would fail. So we filter out ones known by SilFrontendConfig that are relevant for parsing and creating ASTs (which are essentially only the plugin-related ones). I added a comment that explains it.